modules/c++/sys/source/OSUnix.cpp

```diff
@@ -132,13 +133,68 @@ std::set<std::string> get_unique_thread_siblings()
     return unique_ts;
 }
 
-void get_cpu_affinity(cpu_set_t* mask)
+class ScopedCPUAffinity
 {
-    if (-1 == sched_getaffinity(0, sizeof(cpu_set_t), mask))
+public:
+    ScopedCPUAffinity()
     {
-        throw except::Exception(Ctxt("Failed to get CPU affinity"));
+        int numOnlineCPUs = sysconf(_SC_NPROCESSORS_ONLN);
+        if (numOnlineCPUs == -1)
```

Comment:
The current `OSUnix::getNumCPUs()` doesn't do this check (and it should).  Can you move that exception down there and then call the OS method here?  Then change type to `size_t`.